### PR TITLE
[O2B-1138] Implement Data Pass per Simulation Pass view

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- Added Data Passes per Simulation Pass view

Notable changes for developers:
- NA

Changes made to the database:
- NA

### CODE REVIEW

**Comment by xsalonx** [comment]
To be merged after https://github.com/AliceO2Group/Bookkeeping/pull/1370

**Comment by xsalonx** [comment]
To be merged after https://github.com/AliceO2Group/Bookkeeping/pull/1436

File: lib/server/controllers/dataPasses.controller.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ const listDataPassesHandler = async (req, res) => {
     const validatedDTO = await dtoValidator(
         DtoFactory.queryOnly({
             filter: {
+                simulationPassIds: Joi.array().items(Joi.string()),
```
Comment (martinboulais) [inline lib/server/controllers/dataPasses.controller.js:31]: Shouldn't it be an array of numbers?

File: test/public/defaults.js [hunk 2]
```diff
@@ -306,7 +306,21 @@ module.exports.getInnerText = getInnerText;
...
+ * Expect an element to have a text valid against given validator
+ * @param {Object} page Puppeteer page object.
+ * @param {string} selector Css selector.
+ * @param {function<string, boolean>} validator text validator. It must return true if text is valid, retrun false or throw otherwise
+ * @return {Promise<void>} resolves once the text has been checked
+ */
+module.exports.expectInnerTextTo = async (page, selector, validator) => {
+    await page.waitForSelector(selector, { timeout: 200 });
+    const actualInnerText = await getInnerText(await page.$(selector));
+    expect(validator(actualInnerText), `"${actualInnerText}" is invalid with respect of given validator`).to.be.true;
```
Comment (martinboulais) [inline test/public/defaults.js:323]: Nope

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewModel.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+     */
+    get sortModel() {
+        return this._sortModel;
+    }
+
+    /**
+     * Returns data passes names filter model
+     * @return {TextTokensFilterModel} data passes names filter model
+     */
+    get namesFilterModel() {
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewModel.js:124]: applied

File: lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewPage.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+export const DataPassesPerSimulationPassOverviewPage = ({ dataPasses: {
+    perSimulationPassOverviewModel: dataPassesPerSimulationPassOverviewModel } }) => {
+    dataPassesPerSimulationPassOverviewModel.pagination.provideDefaultItemsPerPage(estimateDisplayableRowsCount(
+        TABLEROW_HEIGHT,
+        PAGE_USED_HEIGHT,
+    ));
+
+    const { items, simulationPass, pagination } = dataPassesPerSimulationPassOverviewModel;
+
+    const commonTitle = h('h2', { style: 'white-space: nowrap;' }, 'Data Passes per MC');
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewPage.js]: moved to app.css, when it isn't styled this way, during loading, first part of header is different position than when data are loaded

File: lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewPage.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }, [
+        h('.flex-row.items-center.g2', [
+            filtersPanelPopover(dataPassesPerSimulationPassOverviewModel, dataPassesActiveColumns),
+            h(
+                '.flex-row.g1.items-center',
+                simulationPass.match({
+                    Success: (payload) => breadcrumbs([commonTitle, h('h2', payload.name)]),
+                    Failure: () => [commonTitle, tooltip(h('.f3', iconWarning()), 'Not able to load simulation pass info')],
+                    Loading: () => [commonTitle, spinner({ size: 2, absolute: false })],
+                    NotAsked: () => [commonTitle, tooltip(h('.f3', iconWarning()), 'No data was asked for')],
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewPage.js:56]: simulation pass, it's part of header (data passes Per Simulation Pass)

File: lib/server/controllers/dataPasses.controller.js (hunk 1 above)
Comment (xsalonx) [reply lib/server/controllers/dataPasses.controller.js]: should be, fixed

File: test/public/defaults.js (hunk 2 above)
Comment (xsalonx) [reply test/public/defaults.js]: hmm, some merge artifact, pruned

File: lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewPage.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (martinboulais) [reply lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewPage.js:56]: My bad I read data pass instead of data...

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewPage.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (martinboulais) [reply lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewPage.js]: nowrap is very dangerous, especially on small screen, you should try to find a way to solve your issue using flex configuration instaed

[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewPage.js]: found simple solution, removed

File: lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewModel.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/DataPasses/PerSimulationPassOverview/DataPassesPerSimulationPassOverviewModel.js]: applied, needed to change in DataPassesPerLhcPeriod model

**Review by martinboulais — APPROVED** [review]

**Pull request merged**